modules/logging/src/main/java/com/devonfw/module/logging/common/impl/PerformanceLogFilter.java

```diff
@@ -89,8 +95,11 @@ private void logPerformance(ServletResponse response, long startTime, String url
       errorClass = error.getClass().getName();
       errorMessage = error.getMessage();
     }
-    String message =
-        createMessage(url, Long.toString(duration), Integer.toString(statusCode), errorClass, errorMessage);
+    String correlationId = this.diagnosticContextFacade.getCorrelationId();
+    String message = correlationId != null
+        ? createMessage(correlationId, url, Long.toString(duration), Integer.toString(statusCode), errorClass,
+            errorMessage)
+        : createMessage(url, Long.toString(duration), Integer.toString(statusCode), errorClass, errorMessage);
```

Comment:
This is a missunderstanding. The magic trick of the correlation-id and the MDC is that the logger add this automatically to the log without the logging code line to care about this.
So if it is still present in the MDC it would be logged twice now:
https://github.com/devonfw/devon4j/blob/develop/modules/logging/src/main/java/com/devonfw/module/logging/common/impl/DiagnosticContextFacadeImpl.java#L25

Reply:
Can you confirm? Could you undo this change?

Reply:
Yes. I will take this up tomorrow.

Reply:
I have updated this PR with latest code changes as well as my opening comment to include new use cases.